[DOCU-1841] add content to diff Inso sync and Git-based sync (#19)
Kong/insomnia-docs: docs/insomnia/git-sync.md

---

{:.alert .alert-primary}
**Note**: Sync with Git applies only to Design Documents at this time. Collections use our own Insomnia Sync for versioning. To learn about version control sync for Collections, refer to [Version Control Sync](/insomnia/version-control-sync).

Sync with Git is a built-in feature for Design Documents that enables you to configure your repository to an external Git version control system such as GitHub or BitBucket.

## Enable Git Sync

> Enable Git Sync on local and remote Design Documents by clicking on the **Setup Git Sync** button beside Preferences. Then select **Repository Settings**. A **Configure Repository** modal will open.

Reviewer: There's no concept of local or remote with Documents, only with Projects, which relate to Version Control Sync.

```suggestion
Enable Git Sync on Design Documents by clicking on the **Setup Git Sync** button beside Preferences. Then select **Repository Settings**. A **Configure Repository** modal will open.
```

> ![Enable Git Sync for Documents by clicking the Setup GitSync button inside a Document.](/assets/images/document-git-sync.png)

Reviewer: ```suggestion
![Enable Git Sync for Documents by clicking the Setup Git Sync button inside a Document.](/assets/images/document-git-sync.png)
```

> _Inside a local or remote Design Document, click Setup Git Sync._

Reviewer: ```suggestion
_Inside a Design Document, click Setup Git Sync._
```

## Remote Repository Settings

When configuring a remote repository, you will be prompted for the following information.

> * **Git URI (https)**: The URI of the Git repository. Only HTTPS URLs are supported.

Reviewer: ```suggestion
* **Git URI (https/ssh)**: The URI of the Git repository. Both HTTPS and SSH URLs are supported.
```

Find instructions on how to create a Personal Access Token on the following platforms:

* [Github](https://docs.github.com/en/github/authenticating-to-github/keeping-your-account-and-data-secure/creating-a-personal-access-token)
* [Gitlab](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html)
* [Bitbucket](https://support.atlassian.com/bitbucket-cloud/docs/app-passwords/)
* [Bitbucket Server](https://confluence.atlassian.com/bitbucketserver/personal-access-tokens-939515499.html)

Once complete, click **Done** and the repository settings will be persisted for future operations. Author details and token can be updated as needed.

## Clone an Existing Repository

If a team member has already pushed a Document to a remote repository, it can be cloned via the main menu in the top-right of the application on the Dashboard view. Here, you will see the same Repository Settings dialog to configure remote access.

{:.alert .alert-primary}

> **Note**: Insomnia does not currently support repositories that contain files outside the root **.insomnia** folder.

Reviewer: Funky, this must be from the very first iteration. It _does_ support files outside of the root `.insomnia` folder, it just doesn't consume them (eg. if you have code and your `.insomnia` folder alongside in the same repository, that's fine).

This alert can be removed 🎉
